python/ray/data/_internal/execution/streaming_executor.py

```diff
@@ -439,6 +440,59 @@ def _update_stats_metrics(self, state: str, force_update: bool = False):
             force_update=force_update,
         )
 
+    def _dump_dag_structure(self, dag: PhysicalOperator) -> Dict[str, Any]:
+        """Dump the DAG structure as a dictionary that can be converted to JSON.
+
+        Args:
+            dag: The operator DAG to analyze.
+
+        Returns:
+            A dictionary with the DAG structure, containing operator names,
+            IDs, dependencies, and sub-operators where applicable.
+        """
+        # Return an empty dict if the topology isn't initialized yet
+        if not self._topology:
+            return {}
+
+        # Create a mapping of operators to their IDs
+        op_to_id = {
+            op: self._get_operator_id(op, i) for i, op in enumerate(self._topology)
+        }
+
+        # Create the result structure
+        result = {
+            "operators": [],
+        }
+
+        # Add detailed operator information with dependencies
+        for op in dag.post_order_iter():
+            # Skip operators that aren't in the topology
+            if op not in op_to_id:
+                continue
```

Comment:
For my own understanding, when could this happen?

Reply:
should be safe to remove